src/scratchtocatrobat/converter/converter.py

```diff
@@ -771,6 +933,309 @@ def add_cursor_sprite_to(self, catrobat_scene, upcoming_sprites):
+
+            #length field
+            add_generic_variable_bricks(primary_show_script, primary_hide_script, length_variable, x + VISIBLE_LIST_LEFT_OFFSET + 50, y - (VISIBLE_LIST_HANDLE_WIDTH + num_lines * VISIBLE_LIST_LINE_HEIGHT))
+            handle_show_script.brickList.add(_create_variable_brick(catrobat.create_formula_element_with_value("length"), length_label_variable, catbricks.SetVariableBrick))
+            add_generic_variable_bricks(handle_show_script, handle_hide_script, length_label_variable , x + VISIBLE_LIST_LEFT_OFFSET, y - (VISIBLE_LIST_HANDLE_WIDTH + num_lines * VISIBLE_LIST_LINE_HEIGHT))
+
+            #set offset to 0/length if < 0 or >= length
+            fixed_offset_fe = catformula.FormulaElement(catformula.FormulaElement.ElementType.FUNCTION, str(catformula.Functions.ROUND), None)
+            fixed_offset_fe.setLeftChild(_apply_bounds(_variable_for(offset_variable.getName()), 0, max_offset_fe))
+            copy_loop.loopBricks.add(_create_variable_brick(fixed_offset_fe, offset_variable, catbricks.SetVariableBrick))
+
+            #show handle
+            new_handle_y_fe = catformula.FormulaElement(catformula.FormulaElement.ElementType.OPERATOR, str(catformula.Operators.MINUS), None,
+                                                        catrobat.create_formula_element_with_value(y - VISIBLE_LIST_HANDLE_WIDTH),
+                                                        _get_scaling_fe(_variable_for(offset_variable.getName()), 0, max_offset_fe, 0, height - 2 * VISIBLE_LIST_HANDLE_WIDTH - VISIBLE_VAR_HANDLE_SIZE / 2))
+            show_handle_brick = catbricks.ShowTextColorSizeAlignmentBrick(catformula.Formula(x + VISIBLE_LIST_HANDLE_WIDTH / 2 + 2), catformula.Formula(new_handle_y_fe), catformula.Formula(VISIBLE_VAR_HANDLE_SIZE), catformula.Formula("#000000"))
+            show_handle_brick.setUserVariable(handle_variable)
+            copy_loop.loopBricks.add(show_handle_brick)
+
+            length_fe = catformula.FormulaElement(catformula.FormulaElement.ElementType.FUNCTION, str(catformula.Functions.NUMBER_OF_ITEMS), None, _list_for(name), None)
+            #display variables
+            for i, display_var in enumerate(display_variables):
+                add_generic_variable_bricks(primary_show_script, primary_hide_script, display_var, x + VISIBLE_LIST_LEFT_OFFSET, y - (VISIBLE_LIST_HANDLE_WIDTH + i * VISIBLE_LIST_LINE_HEIGHT))
+                item_offset_fe = catformula.FormulaElement(catformula.FormulaElement.ElementType.OPERATOR, str(catformula.Operators.PLUS), None,
+                                                           _variable_for(offset_variable.getName()),
+                                                           catrobat.create_formula_element_with_value(i+1))
+                variable_value_fe = catformula.FormulaElement(catformula.FormulaElement.ElementType.FUNCTION, str(catformula.Functions.JOIN), None,
```

Comment:
The creation of formula elements in this part of the code make it really hard to read. Maybe some more intermediate variables would help. Or even better would be to have separate methods creating some of the more complex formulas like this one.

Reply:
As soon as #171 is merged I'll rewrite the code to use the formula_element_for function to improve the readability. The formula elements that then still look complex I'll move to own functions.